gobblin-modules/gobblin-compliance/src/main/java/gobblin/compliance/purger/HivePurgerPublisher.java

```diff
@@ -41,12 +76,47 @@ public void publishData(Collection<? extends WorkUnitState> states) {
     for (WorkUnitState state : states) {
       if (state.getWorkingState() == WorkUnitState.WorkingState.SUCCESSFUL) {
         state.setWorkingState(WorkUnitState.WorkingState.COMMITTED);
+        submitEvent(state, ComplianceEvents.Purger.WORKUNIT_COMMITTED);
       } else {
         state.setWorkingState(WorkUnitState.WorkingState.FAILED);
+        submitEvent(state, ComplianceEvents.Purger.WORKUNIT_FAILED);
       }
     }
   }
 
+  private void submitEvent(WorkUnitState state, String name) {
+    WorkUnit workUnit = WorkUnit.createEmpty();
+    workUnit.addAll(state.getWorkunit());
+    workUnit.setProp(ComplianceConfigurationKeys.WORKUNIT_RECORDSREAD,
+        workUnit.getProp(ComplianceConfigurationKeys.NUM_ROWS));
+    workUnit.setProp(ComplianceConfigurationKeys.WORKUNIT_BYTESREAD, getDataSize(workUnit));
+    Optional<HivePartitionDataset> dataset =
+        HivePurgerSource.findDataset(workUnit.getProp(ComplianceConfigurationKeys.PARTITION_NAME), this.datasets);
+    if (!dataset.isPresent()) {
+      return;
+    }
+    HivePartitionDataset hivePartitionDataset = dataset.get();
+    HivePurgerSource.setNumRowsInWorkUnit(hivePartitionDataset, workUnit);
```

Comment:
It might be better to introduce a HivePartitionDataset.getParam(String name) method so that this reads:

numRows = partition.getParam(Keys.NUM_ROWS);
rawSize = partition.getParam(Keys.RAW_SIZE);
workunit.setProp(Keys.BYTES_READ, numRows);
workunit.setProp(Keys.RAW_SIZE, rawSize);